src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlStringMethodTranslator.cs

```diff
@@ -243,6 +257,28 @@ public SqlExpression Translate(SqlExpression instance, MethodInfo method, IReadO
             if (method == EndsWith)
                 return TranslateStartsEndsWith(instance, arguments[0], false);
 
+            if (method == JoinStringIEnumerable ||
+                method == JoinString || method == JoinStringCharSep ||
+                method.IsClosedFormOf(JoinGeneric) || method.IsClosedFormOf(JoinGenericCharSep))
+            {
+                var array = arguments[1];
+                if (!array.Type.TryGetElementType(out var operandElementType))
+                    return null;
+
+                if (array.TypeMapping is RelationalTypeMapping typeMapping &&
+                    !(typeMapping is NpgsqlArrayTypeMapping) && !(typeMapping is NpgsqlJsonTypeMapping))
+                {
+                    return null;
+                }
+
+                return _sqlExpressionFactory.Function(
+                    "array_to_string",
+                    new[] { array, arguments[0], _sqlExpressionFactory.Constant("") },
+                    nullable: true,
+                    argumentsPropagateNullability: new[] { true, true, false },
```

Comment:
The 3rd parameter does propagate null, even though in your particular translation it is always a non-null constant. For clarity/consistency, best to just put TrueArray[3].